dali_tf_plugin/setup.py.in

```diff
@@ -0,0 +1,203 @@
+            tf_lflags=" ".join(["-L" + tensorflow_path, "-ltensorflow_framework"])
+
+    if tf_cflags is '' and tf_lflags is '':
+        raise ImportError('Could not find Tensorflow. Tensorflow must be installed before installing nvidia-dali-tf-plugin')
+    return (tf_cflags, tf_lflags)
+
+def get_tf_compiler_version():
+    tensorflow_path = get_module_path('tensorflow')
+    tensorflow_libs = find('libtensorflow_framework*so*', tensorflow_path)
+    if not tensorflow_libs:
+        return ''
+    lib = tensorflow_libs[0]
+    cmd = 'strings -a ' + lib + ' | grep "GCC: ("'
+    s = str(subprocess.check_output(cmd, shell=True))
+    version = re.search("GCC:\s*\(.*\)\s*(\d+.\d+).\d+", s).group(1)
+    return version
+
+def get_tf_version():
+    import pkg_resources
+    s = pkg_resources.get_distribution("tensorflow-gpu").version
+    version = re.search("(\d+.\d+).\d+", s).group(1)
+    return version
+
+def get_gcc_compiler_version():
+    cmd = 'gcc --version | grep "gcc ("'
+    s = str(subprocess.check_output(cmd, shell=True))
+    version = re.search("gcc\s*\(.*\)\s*(\d+.\d+).\d+", s).group(1)
+    return version
+
+def find_available_prebuilt_tf(requested_version, available_libs):
+    req_ver_first, req_ver_second = [int(v) for v in requested_version.split('.', 2)]
+    selected_ver = None
+    for file in available_libs:
+        re_match = re.search(".*(\d+)_(\d+).*", file)
+        if re_match is None:
+            continue
+        ver_first, ver_second = [int(v) for v in re_match.groups()]
+        if ver_first == req_ver_first:
+            if ver_second <= req_ver_second and (selected_ver is None or selected_ver < (ver_first, ver_second)):
+                selected_ver = (ver_first, ver_second)
+    return '.'.join([str(v) for v in selected_ver]) if selected_ver is not None else None
+
+def get_dali_build_flags():
+    dali_cflags = ''
+    dali_lflags = ''
+    try:
+        import nvidia.dali.sysconfig as dali_sc
+        dali_lib_path = dali_sc.get_lib_dir()
+        dali_cflags=" ".join(dali_sc.get_compile_flags())
+        dali_lflags=" ".join(dali_sc.get_link_flags())
+    except:
+        dali_path = get_module_path('nvidia/dali')
+        if dali_path is not '':
+            dali_cflags=" ".join(["-I" + dali_path + "/include", "-D_GLIBCXX_USE_CXX11_ABI=0"])
+            dali_lflags=" ".join(["-L" + dali_path, "-ldali"])
+    if dali_cflags is '' and dali_lflags is '':
+        raise ImportError('Could not find DALI.')
+    return (dali_cflags, dali_lflags)
+
+def get_cuda_build_flags():
+    cuda_cflags = ''
+    cuda_lflags = ''
+    cuda_home = os.environ.get('CUDA_HOME')
+    if not cuda_home:
+        cuda_home = '/usr/local/cuda'
+    cuda_cflags=" ".join(["-I" + cuda_home + "/include"])
+    cuda_lflags=" ".join([])
+    return (cuda_cflags, cuda_lflags)
+
+def which(program):
+    def is_exe(fpath):
```

Comment:
Can't we just call system `which` command?

Reply:
I think it's better to use python than invoking another tool and then parse the standard output. If you think it's better to use which command I can try to rework it